src/neo/Ledger/Blockchain.cs

```diff
@@ -320,6 +321,7 @@ private void Persist(Block block)
                         if (engine.Execute() == VMState.HALT)
                         {
                             clonedSnapshot.Commit();
+                            if (!isGasCalled && engine.TouchedContracts.Contains(NativeContract.GAS.Hash)) isGasCalled = true;
```

Comment:
It will always be `true` because of fees.

Reply:
But fees doesn't call `Invoke`, so it won't be in touched contracts, only if they call `Transfer`, `OnPersist` don't

Reply:
@neo-project/ngd-shanghai could you test it?

Reply:
for example, transfer only txs? or invoke some contracts txs?

Reply:
Yes, transfer neo or contract invocations